sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/resolver/TimezoneAwareExpressionResolver.scala

```diff
@@ -58,12 +59,14 @@ class TimezoneAwareExpressionResolver(expressionResolver: ExpressionResolver)
       expressionWithResolvedChildren,
       traversals.current.sessionLocalTimeZone
     )
+
     coerceExpressionTypes(
       expression = expressionWithResolvedChildrenAndTimeZone,
-      expressionTreeTraversal = traversals.current
-    ) match {
+      expressionTreeTraversal = traversals.current) match {
       case cast: Cast if traversals.current.defaultCollation.isDefined =>
         tryCollapseCast(cast, traversals.current.defaultCollation.get)
+      case Cast(child, TimestampNTZType, _, _) if child.dataType.isInstanceOf[TimeType] =>
+        MakeTimestampNTZ(CurrentDate(), child)
```

Comment:
Is it really time zone aware expression? How about to create a separate resolver.